pkg/daemon/ceph/mds/init.go

```diff
@@ -57,31 +55,25 @@ type Config struct {
 func Initialize(context *clusterd.Context, config *Config) error {
 	logger.Infof("Creating config for mds %s [filesystem ID: %s] [active standby?: %+v]",
 		config.Name, config.FilesystemID, config.ActiveStandby)
-	err := generateConfigFiles(context, config)
-	if err != nil {
-		return fmt.Errorf("failed to generate mds config files: %+v", err)
-	}
+	config.ClusterInfo.Log(logger)
 
-	util.WriteFileToLog(logger, cephconfig.DefaultConfigFilePath())
-
-	return err
-}
-
-func generateConfigFiles(context *clusterd.Context, config *Config) error {
 	// write the latest config to the config dir
+	// TODO: what is this necessary for? I don't think the mds uses the config/keyring generated here.
```

Comment:
What is this used for? It doesn't seem like it has a purpose to me, but I could be missing something.

Reply:
you can probably remove the admin key generation since the mds key is used. It'll break pretty quickly if it is actually needed somewhere though.